src/main/java/org/candlepin/insights/controller/InventoryController.java

```diff
@@ -41,65 +42,91 @@
 
-        String ipAddresses = consumer.getFacts().get("ip-addresses");
-        if (ipAddresses != null) {
-            String[] ipAddressesSplit = ipAddresses.split(", ");
-            facts.setIpAddresses(Arrays.asList(ipAddressesSplit));
-        }
+        List<String> productIds = consumer.getInstalledProducts().stream()
+            .map(installedProduct -> installedProduct.getProductId().toString()).collect(Collectors.toList());
+        facts.setRhProd(productIds);
 
-        facts.setFqdn(consumer.getFacts().get("network.fqdn"));
+        return facts;
+    }
 
-        String macAddresses = consumer.getFacts().get("mac-addresses");
-        if (macAddresses != null) {
-            String[] macAddressesSplit = macAddresses.split(", ");
-            facts.setMacAddresses(Arrays.asList(macAddressesSplit));
+    private void extractHardwareFacts(Map<String, String> pinheadFacts, ConduitFacts facts) {
+        String systemUuid = pinheadFacts.get("dmi.system.uuid");
+        if (!isEmpty(systemUuid)) {
+            facts.setBiosUuid(UUID.fromString(systemUuid));
         }
 
-        String cpuSockets = consumer.getFacts().get("cpu.cpu_socket(s)");
+        String cpuSockets = pinheadFacts.get("cpu.cpu_socket(s)");
         Integer numCpuSockets = null;
-        if (cpuSockets != null) {
+        if (!isEmpty(cpuSockets)) {
             numCpuSockets = Integer.parseInt(cpuSockets);
             facts.setCpuSockets(numCpuSockets);
         }
 
-        String coresPerSocket = consumer.getFacts().get("cpu.core(s)_per_socket");
-        if (coresPerSocket != null && cpuSockets != null) {
+        String coresPerSocket = pinheadFacts.get("cpu.core(s)_per_socket");
+        if (!isEmpty(coresPerSocket) && !isEmpty(cpuSockets)) {
             Integer numCoresPerSocket = Integer.parseInt(coresPerSocket);
             facts.setCpuCores(numCoresPerSocket * numCpuSockets);
         }
 
-        String memoryTotal = consumer.getFacts().get("memory.memtotal");
-        if (memoryTotal != null) {
+        String memoryTotal = pinheadFacts.get("memory.memtotal");
+        if (!isEmpty(memoryTotal)) {
             int memoryBytes = Integer.parseInt(memoryTotal);
             // memtotal is a little less than accessible memory, round up to next GB
             int memoryGigabytes = (int) Math.ceil((float) memoryBytes / (float) KIBIBYTES_PER_GIBIBYTE);
             facts.setMemory(memoryGigabytes);
         }
 
-        facts.setArchitecture(consumer.getFacts().get("uname.machine"));
+        String architecture = pinheadFacts.get("uname.machine");
+        if (!isEmpty(architecture)) {
+            facts.setArchitecture(architecture);
+        }
+    }
+
+    private void extractNetworkFacts(Map<String, String> pinheadFacts, ConduitFacts facts) {
+        String ipAddresses = pinheadFacts.get("Ip-addresses");
+        if (!isEmpty(ipAddresses)) {
+            String[] ipAddressesSplit = ipAddresses.split(", ");
```

Comment:
Let's be a little more robust and split on ",\s*" so the space becomes optional and multiple spaces are eaten by the regex engine